test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -2710,83 +2665,46 @@ func TestSchedulingGatesPluginEventsToRegister(t *testing.T) {
 			)
 			defer teardown()
 
-			// Create a pod with schedulingGates.
-			gatedPod := st.MakePod().Name("p").Namespace(testContext.NS.Name).
-				SchedulingGates([]string{"foo"}).
+			// Create a pause pod.
+			pausePod := st.MakePod().Name("pause").Namespace(testContext.NS.Name).
 				PodAffinity("kubernetes.io/hostname", &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}, st.PodAffinityWithRequiredReq).
 				Container("pause").Obj()
-			gatedPod, err := testutils.CreatePausePod(testCtx.ClientSet, gatedPod)
+			pausePod, err := testutils.CreatePausePod(testCtx.ClientSet, pausePod)
 			if err != nil {
 				t.Errorf("Error while creating a gated pod: %v", err)
 				return
 			}
 
-			if err := testutils.WaitForPodSchedulingGated(testCtx.ClientSet, gatedPod, 10*time.Second); err != nil {
+			if err := testutils.WaitForPodUnschedulableWithTimeout(testCtx.ClientSet, pausePod, 10*time.Second); err != nil {
 				t.Errorf("Expected the pod to be gated, but got: %v", err)
 				return
 			}
-			if num(tt.enqueuePlugin) != 1 {
-				t.Errorf("Expected the preEnqueue plugin to be called once, but got %v", num(tt.enqueuePlugin))
+			// Run twice for Filter logics in scheduler.
+			if tt.plugin.(*FilterPlugin).numFilterCalled != 2 {
+				t.Errorf("Expected the plugin to be called once, but got %v", tt.plugin.(*FilterPlugin).numFilterCalled)
```

Comment:
```suggestion
				t.Errorf("Expected the plugin to be called twice, but got %v", tt.plugin.(*FilterPlugin).numFilterCalled)
```

But also, why is it 2?

Reply:
Because we have two nodes, updated the comments.